bot/seasons/evergreen/battleship.py

```diff
@@ -107,211 +111,227 @@ def __init__(
-            await self.public_channel.send(f"{self.player2}'s Board:\n{self_grid_2}")
+        await self.public_channel.send(f"Game Over! {winner.mention} won against {loser.mention}")
 
-        self.bot.get_cog("Battleship").games.remove(self)
+        for player in (self.p1, self.p2):
+            grid = self.format_grid(player, SHIP_EMOJIS)
+            await self.public_channel.send(f"{player.user}'s Board:\n{grid}")
 
     @staticmethod
-    def check_sink(grid: grid_typehint, boat: str) -> bool:
+    def check_sink(grid: Grid, boat: str) -> bool:
         """Checks if all squares containing a given boat have sunk."""
         return all(square.aimed for row in grid for square in row if square.boat == boat)
 
     @staticmethod
-    def check_gameover(grid: grid_typehint) -> bool:
+    def check_gameover(grid: Grid) -> bool:
         """Checks if all boats have been sunk."""
         return all(square.aimed for row in grid for square in row if square.boat)
 
     def setup_grids(self) -> None:
         """Places the boats on the grids to initialise the game."""
-        for player in (self.player1, self.player2):
-            for name, size in ships.items():
+        for player in (self.p1, self.p2):
+            for name, size in SHIPS.items():
                 while True:  # Repeats if about to overwrite another boat
                     overwrite = False
                     coords = []
+
+                    coord1 = random.randint(0, 9)
+                    coord2 = random.randint(0, 10 - size)
+
                     if random.choice((True, False)):  # Vertical or Horizontal
-                        # Vertical
-                        x_coord = random.randint(0, 9)
-                        y_coord = random.randint(0, 9 - size)
-                        for i in range(size):
-                            if self.grids[player][x_coord][y_coord + i].boat:  # Check if there's already a boat
-                                overwrite = True
-                            coords.append((x_coord, y_coord + i))
+                        x, y = coord1, coord2
+                        xincr, yincr = 0, 1
                     else:
-                        # Horizontal
-                        x_coord = random.randint(0, 9 - size)
-                        y_coord = random.randint(0, 9)
-                        for i in range(size):
-                            if self.grids[player][x_coord + i][y_coord].boat:  # Check if there's already a boat
-                                overwrite = True
-                            coords.append((x_coord + i, y_coord))
+                        x, y = coord2, coord1
+                        xincr, yincr = 1, 0
+
+                    for i in range(size):
+                        new_x = x + (xincr * i)
+                        new_y = y + (yincr * i)
+                        if player.grid[new_x][new_y].boat:  # Check if there's already a boat
+                            overwrite = True
```

Comment:
Should break out of the loop since there's no point in continuing - the `coords` list will get reset anyway. Another way to implement this control flow would be 

```py
overwrite = True
while overwrite:
    ...

    for i in range(size):
        ...
        
        if player.grid[new_x][new_y].boat:
            break
        coords.append((new_x, new_y))
    else
        overwrite = False
```

but you can stick with the infinite loop if you prefer.